pkg/nsx/services/subnetipreservation/cleanup.go

```diff
@@ -0,0 +1,28 @@
+package subnetipreservation
+
+import (
+	"context"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+func (s *IPReservationService) CleanupBeforeVPCDeletion(ctx context.Context) error {
```

Comment:
could you also add unit test for cleanup?

Reply:
Added, thanks